libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -31,65 +31,69 @@
 
 /* Test includes */
 #include <string.h>
+
+/* Test includes */
 #include "unity.h"
 #include "unity_fixture.h"
 
 /* Driver includes */
 #include "iot_spi.h"
+
+/* FreeRTOS includes */
 #include "FreeRTOS.h"
 #include "semphr.h"
 
 #define SPI_BUFFER_SIZE (32)
-#define testIotSPI_DEFAULT_SEMAPHORE_DELAY (100000U)
-#define SPI_TEST_FREQ (500000U)
+#define testIotSPI_DEFAULT_SEMAPHORE_DELAY (3000U)
 
 /*-----------------------------------------------------------*/
 /* Globals values which can be overwritten by the test
  * framework invoking these tests */
 /*-----------------------------------------------------------*/
-uint32_t ultestIotSpiInstance = 3;                            /* Use SPI instance 0 */
-IotSPIMode_t xtestDefaultConfigMode = eSPIMode0;              /* Use SPI Mode eSPIMode0 */
-IotSPIBitOrder_t xtestDefaultconfigBitOrder = eSPIMSBFirst;   /* Use SPI Bit Order eSPIMSBFirst */
-static SemaphoreHandle_t xtestIotSpiSemaphore = NULL;
-
+uint32_t ultestIotSpiInstance = 0;                            /* Test SPI Instance */
+uint32_t ultestIotSPIFrequency = 500000U;                     /* Test SPI Frequency */
+uint32_t ultestIotSPIDummyValue = 0;                          /* Test SPI Dummy Value */
+IotSPIMode_t xtestIotSPIDefaultConfigMode = eSPIMode0;              /* Default SPI eSPIMode0 */
+IotSPIBitOrder_t xtestIotSPIDefaultconfigBitOrder = eSPIMSBFirst;   /* Default SPI bit order eSPIMSBFirst */
+static SemaphoreHandle_t xtestIotSPISemaphore = NULL;
+static StaticSemaphore_t xtestIotSPICompleted;
 /*-----------------------------------------------------------*/
 
-static void prvSpiAsyncCallback(void *pvUserContext)
+/**
+ * @brief Application/POSIX defined callback for asynchronous operations
+ * This callback function releases a semaphore every time it is called.
+ */
+static void prvSpiAsyncCallback( IotSPITransactionStatus_t xStatus, void *pvUserContext )
 {
     BaseType_t xHigherPriorityTaskWoken;
 
-    xSemaphoreGiveFromISR(xtestIotSpiSemaphore, &xHigherPriorityTaskWoken);
+    xSemaphoreGiveFromISR(xtestIotSPISemaphore, &xHigherPriorityTaskWoken);
     portYIELD_FROM_ISR(xHigherPriorityTaskWoken);
 }
-
-
-
-
+/*-----------------------------------------------------------*/
 
 /*-----------------------------------------------------------*/
 /* Define Test Group. */
 TEST_GROUP( TEST_IOT_SPI );
-
 /*-----------------------------------------------------------*/
 
 /**
  * @brief Setup function called before each test in this group is executed.
  */
 TEST_SETUP( TEST_IOT_SPI )
 {
-    xtestIotSpiSemaphore = xSemaphoreCreateBinary();
-    TEST_ASSERT_NOT_EQUAL(NULL, xtestIotSpiSemaphore);
+   xtestIotSPISemaphore = xSemaphoreCreateCountingStatic( 10, 0, &xtestIotSPICompleted );
```

Comment:
Whats the advantage of using counting here?

Reply:
I will revert this to binary